Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -306,6 +307,34 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Virtual method - must be overridden by concrete algorithm
   virtual void exec() = 0;
 
+  virtual void execDistributed() { exec(); }
+  virtual void execMasterOnly();
+  virtual void execNonMaster(){};
```

Comment:
I'm sure there's a logical explanation for this: How is non-master different from distributed?